Add visual-editing package (#1)
directus/visual-editing: readme.md

        website)
      - `onSaved`: Optional callback function `(data) => void`. Will be called after the fields are saved in Directus.
        If `onSaved` is not set, the current page will be reloaded by using `window.location.reload();`
        - `onSaved` parameters: `{ collection, item, payload }`
          - `collection`: `string`
          - `item`: `PrimaryKey` (`string` | `number`)
          - `payload`: `Record<string, any>` (the changed values)
    - Returns: `{ remove, enable, disable }` … for all selected elements you can use these methods
      - Make sure to `await` them
        ```ts
        const { remove, enable, disable } = await scan({ directusUrl });
        ```
        - `remove()` … remove the previously scanned elements
        - `disable()` … disable the previously scanned elements
        - `enable()` enable the previously _disabled_ elements
  - `remove` … removes all elements
    - Before leaving the page on client-side navigation, be sure to call the `remove` method
  - `disable` … disable all elements (i.e. they are temporarily not editable)
    - Returns: `{ enable }`
      - `enable()` … will enable all elements previously _disabled_

> - If you have CSP configured be sure to add `'frame-ancestors': ["'self'", process.env.DIRECTUS_URL],` … in other words:
>   make it available for use inside an iframe in Directus.

Reviewer: ```suggestion
- If you have CSP configured, be sure to add 'frame-ancestors': ["'self'", process.env.DIRECTUS_URL] to your policy — in other words, make your site available for use inside an iframe in Directus.
```

> 

Reviewer: Maybe a bit more context regarding CSP would be an improvement:

"If you're unsure where your CSP is defined, check your web server configuration files, your site's build configuration, or your hosting platform's security settings."

Author: Makes sense! Ty!

## Usage for Non-JS environments

For server-only rendered websites (like PHP)

- You can find the compiled version in [dist/visual-editing.js](dist/visual-editing.js)
- Include the package in your website
  ```html
  <script src="path-to/visual-editing.js" type="text/javascript"></script>
  ```
- Usage
  ```html
  <script type="text/javascript">
  	document.addEventListener('DOMContentLoaded', function () {
  		DirectusVisualEditing.scan({ directusUrl: 'http://localhost:8000' });
  	});
  </script>
  ```

## Rendering Modes

> - Note to DX: Take a look at the test website to see how to deal with rendering modes.

Reviewer: ```suggestion
- Note to DX: Take a look at the [test website's README](test-website/readme.md)
```

Author: As mentioned, this is just a note to the team working on the docs.

> [!NOTE]  
> This document is WIP

# Directus Visual Editing Package

## Installation & Usage

- Install the package: `pnpm i @directus/visual-editing`
- Usage
  ```ts
  import { scan, remove, disable, toEditAttr } from '@directus/visual-editing';
  ```
- Add `data-directus` attributes to your elements and then call the `scan()` method to make them interactive

>   - the `data-directus` content follows this syntax `collection:posts;item:12` or

Reviewer 2: Instead of "Content follows this syntax `collection:posts;item:12`" it might be better to make it clearer like `collection:<collection_name>;item:<item_key>` and then the example

    `collection:posts;item:12;fields:title,description;mode:drawer` … described in detail below
  - It is recommended to render only the `data-directus` attributes when using in Visual Editor. This can be done in
    several ways. This could be by adding query parameters like `?visual-editing=true&token=123` or something else that
    you would implement on your website. The reason is that it could expose data like primary keys that you might not

>     want to show to the public.

Reviewer 2: ```suggestion
  - It is recommended to render only the `data-directus` attributes outlined in this readme when setting up your site for use with the Visual Editor. For example, DO NOT add query parameters like `?visual-editing=true&token=123` or any other sensitive data. Doing so could expose data like primary keys that you might not want to show to the public.
```

This might be better phrasing, or something along these lines.

Author: Well, your suggestion turns the content into the opposite. Updated the sentence for clarity.

Reviewer 2: > Well, your suggestion turns the content into the opposite

Well clearly I didn't understand what you were trying to say then, which I think proves the point 🫨

>     could expose data like primary keys that you might not want to show to the public.

Reviewer 2: I'm still struggling to understand what you're trying to say here

Author: Let’s discuss this in our meeting.
